cmd/juju/cloud/detectcredentials.go

```diff
@@ -147,6 +147,8 @@ func (c *detectCredentialsCommand) Run(ctxt *cmd.Context) error {
 		defaultCloudNames[cloud.Type] = cloudName
 	}
 
+	// get all provider Names that are registered
```

Comment:
I believe the function names "registeredProvidersFunc" and "sort.Strings" explain well enough what they do. The additional comment is not helpful, please revert